cloudformation/auxiliary_templates/custom_balancer/custom_queue_job_balancer.yaml

```diff
@@ -0,0 +1,401 @@
+                              "operator": "EQUAL",
+                              "value": "RUNNING",
+                          }
+                      },
+                  ],
+                  "operator": "OR",
+              }
+          
+              job_search_response = deadline.search_jobs(
+                  filterExpressions=filter_expressions,
+                  farmId=farm_id,
+                  queueIds=[queue_id],
+                  itemOffset=0,
+                  pageSize=100,
+              )
+              active_jobs.extend(job_search_response["jobs"])
+          
+              while job_search_response.get("nextItemOffset"):
+                  job_search_response = deadline.search_jobs(
+                      filterExpressions=filter_expressions,
+                      farmId=farm_id,
+                      queueIds=[queue_id],
+                      itemOffset=job_search_response.get("nextItemOffset"),
+                      pageSize=100,
+                  )
+                  active_jobs.extend(job_search_response["jobs"])
+              return active_jobs
+          
+          
+          def get_all_associated_fleets(deadline, farm_id, queue_id):
+              associated_fleets = []
+              qfa_paginator = deadline.get_paginator("list_queue_fleet_associations")
+              for qfa_page in qfa_paginator.paginate(farmId=farm_id, queueId=queue_id):
+                  associated_fleets.extend(
+                      [
+                          x["fleetId"]
+                          for x in qfa_page.get("queueFleetAssociations", [])
+                          if x["status"] == "ACTIVE"
+                      ]
+                  )
+              return associated_fleets
+          
+          
+          def get_worker_limit(deadline, farm_id, fleet_id):
+              fleet_info = deadline.get_fleet(farmId=farm_id, fleetId=fleet_id)
+              return fleet_info["maxWorkerCount"]
+          
+          
+          def calculate_job_weights(deadline, farm_id, queue_id, job):
+              # get every step
+              steps = []
+          
+              steps_paginator = deadline.get_paginator("list_steps")
+              for step_page in steps_paginator.paginate(farmId=farm_id, queueId=queue_id, jobId=job["jobId"]):
+                  steps.extend(step_page.get("steps", []))
+          
+              # get every task
+              tasks = []
+              for step in steps:
+                  tasks_paginator = deadline.get_paginator("list_tasks")
+                  for task_page in tasks_paginator.paginate(farmId=farm_id, queueId=queue_id, jobId=job["jobId"], stepId=step["stepId"]):
+                      tasks.extend(task_page.get("tasks", []))
+          
+              error_count = sum([x["failureRetryCount"] for x in tasks])
+              # total_rendering_tasks = sum([x["runStatus"] in ["RUNNING", "ASSIGNED", "STARTING", "SCHEDULED", "READY"] for x in tasks])
```

Comment:
I believe we do not need to iterate over all tasks here. Jobs have [taskRunStatusCounts](https://docs.aws.amazon.com/deadline-cloud/latest/APIReference/API_GetJob.html#deadlinecloud-GetJob-response-taskRunStatusCounts), so it'd be easier just to use those counts for calculation.

Also, why do we need this comment? It seems to be the same as code before, just formatted differently, but includes `READY` status.